feat: JPA 엔티티 생성

dhfkdlsj opened the pull request:
## 개요
- JPA 엔티티 생성
- 생성, 수정 시간을 저장하는 엔티티 생성

Resolves: #15 (JPA 엔티티 생성)

## PR 유형
어떤 변경 사항이 있나요?

- [x] 새로운 기능 추가
- [ ] 버그 수정
- [ ] CSS 등 사용자 UI 디자인 변경
- [ ] 코드에 영향을 주지 않는 변경사항(오타 수정, 탭 사이즈 변경, 변수명 변경)
- [ ] 코드 리팩토링
- [ ] 주석 추가 및 수정
- [ ] 문서 수정
- [ ] 테스트 추가, 테스트 리팩토링
- [ ] 빌드 부분 혹은 패키지 매니저 수정
- [x] 파일 혹은 폴더명 수정
- [ ] 파일 혹은 폴더 삭제

## PR Checklist
PR이 다음 요구 사항을 충족하는지 확인하세요.

- [x] 커밋 메시지 컨벤션에 맞게 작성했습니다.
- [x] 변경 사항에 대한 테스트를 했습니다.(버그 수정/기능에 대한 테스트).

dhfkdlsj committed:
Feat: BaseEntity 생성

- 생성/수정 시간 필드를 관심사에 따라 분리
- 불필요한 상속 방지를 위한 CreatedBaseEntity, TimestampedEntity 구조 도입

Resolves: #15

dhfkdlsj committed:
Feat: Post Entity 생성

- Post Entity 생성
- Category는 Enum 타입으로 관리

Resolves: #15

dhfkdlsj committed:
Chore: Enum 패키지명을 enums 로 변경

- Enum 패키지명을 enums 로 변경

Resolves: #15

dhfkdlsj committed:
Feat: Comment 엔티티 추가

- Comment 엔티티 추가

Resolves: #15

dhfkdlsj committed:
Feat: Like 엔티티 추가

- Like 엔티티 추가

Resolves: #15

dhfkdlsj committed:
Feat: Member 엔티티 추가

- Member 엔티티 추가
- 로그인 방식을 Enum 타입으로 관리
- 멤버의 Role을 Enum 타입으로 관리

Resolves: #15

dhfkdlsj committed:
Feat: Organization 엔티티 추가

- Organization 엔티티 추가
- Enum타입 Corporation을 WorkPlace와 Organization에서 사용하기 때문에 global에 추가

Resolves: #15

dhfkdlsj committed:
Feat: WorkPlace 엔티티 추가

- WorkPlace 엔티티 추가

Resolves: #15

dhfkdlsj committed:
Feat: Project 엔티티 추가

- Project 엔티티 추가

Resolves: #15

dhfkdlsj committed:
Chore: baseentity 패키지 네임 변경

- global의 baseentity에 Enum타입 Corporation이 추가되었으므로 패키지 네임을 baseentity에서 entity로 변경

Resolves: #15

dhfkdlsj committed:
Chore: Code Style 변경

- Code Style을 Google로 변경

Resolves: #15

dhfkdlsj committed:
Feat: @EnableJpaAuditing 추가

- JPA Auditing 기능 활성화를 위해 @EnableJpaAuditing 추가

Resolves: #15

HaegyeongKim01 reviewed (commented):
굿! 확인했습니다 👍

heodongwon commented on src/main/java/com/itcen/censquare/domain/post/entity/Post.java:
```diff
@@ -1,4 +1,59 @@
+  private Long viewCount = 0L;
+
+  @Column(name = "like_count")
+  private Long likeCount = 0L;
```
Builder.Default 안달아도 되나요 3트

dhfkdlsj replied:
빌더로 객체를 생성할 때 해당 필드에 값을 명시적으로 넣어주려고 했기 때문에 처음에는 @Builder.Default를 사용하지 않았습니다.
하지만 혹시 모를 값 누락이나 실수를 방지하기 위해 바로 @Builder.Default를 추가하겠습니다!

dhfkdlsj committed:
Feat: @Builder.Default 추가

- 빌더 사용 시 null 대신 기본값 유지하도록 @Builder.Default 추가
- viewCount, likeCount 필드에 기본값 0L 보장
- 필드 누락 시도 방지 및 객체 생성 안정성 확보

Resolves: #15

dhfkdlsj committed:
Feat: @NoArgsConstructor에 access = AccessLevel.PROTECTED 설정 추가

- JPA 프록시 생성을 위해 기본 생성자는 필요하되, 외부에서 직접 사용하지 못하도록 보호 수준 설정
- @NoArgsConstructor(access = AccessLevel.PROTECTED) 적용하여 불완전한 객체 생성을 방지

Resolves: #15

dhfkdlsj committed:
Refactor: Like 엔티티명을 PostLike로 변경

- SQL 예약어인 LIKE와의 충돌로 인해 엔티티명과 테이블명을 PostLike로 변경
- PostgreSQL에서 DDL 실행 오류 발생 (syntax error at or near "Like") 대응
- 클래스명, 테이블명 일괄 수정하여 명확성 및 호환성 확보

Resolves: #15

heodongwon reviewed (approved):
LGTM
